dashboard/src/actions/profileActions.js

```diff
@@ -35,31 +34,29 @@ export const getProfileDetails = () => async (dispatch) => {
 export const updateUserDetails =
   (value, fieldName) => async (dispatch, getState) => {
     let userDetails = { ...getState().userProfile.userDetails };
-    userDetails[fieldName] = value;
+    let updatedUserDetails = { ...getState().userProfile.updatedUserDetails };
 
+    userDetails[fieldName] = value;
+    updatedUserDetails[fieldName] = value;
+    let payload = {
+      userDetails, 
+      updatedUserDetails
+    }
     dispatch({
       type: TYPES.UPDATE_USER_DETAILS,
-      payload: { ...userDetails },
+      payload
     });
   };
 
-export const sendForUpdation = () => async (dispatch, getState) => {
+export const sendForUpdate = () => async (dispatch, getState) => {
   try {
     dispatch({ type: TYPES.LOADING });
-    const userDetails = { ...getState().userProfile.userDetails };
-    const userDetails_copy = { ...getState().userProfile.userDetails_copy };
-
-    let keys = ["first_name", "last_name", "email"];
-    let params = {};
-    for (let key of keys) {
-      if (userDetails[key] !== userDetails_copy[key]) {
-        params[key] = userDetails[key];
-      }
-    }
+    const updatedUserDetails = { ...getState().userProfile.updatedUserDetails };
+        
     const username = getState().userAuth.loginDetails.username;
     if (username) {
       const response = await API.put(`${API_ROUTES.GET_USER}/${username}`, {
-        ...params,
+        ...updatedUserDetails,
```

Comment:
We also need to check that `updatedUserDetails` is not empty.

Reply:
Only if `updatedUserDetails` not empty, `Save` button will be enabled and request will be sent. Else-clause should be handled